src/js/components/calendar.jsx

```diff
@@ -108,21 +153,59 @@ export default class Calendar extends React.Component {
     });
   }
 
-  _onTimeslotSelected(timeslot) {
+  _onTimeslotClick(newTimeslot) {
+    const {
+      selectedTimeslots,
+    } = this.state;
+
+    const {
+      maxTimeslots,
+    } = this.props;
+
+    const newSelectedTimeslots = selectedTimeslots.slice();
+
+    let existentTimeslotIndex = -1;
+    const timeslotExists = newSelectedTimeslots.some((timeslot, index) => {
+      existentTimeslotIndex = index;
+      return newTimeslot.startDate.format() === timeslot.startDate.format();
+    });
+
+    if (timeslotExists) {
+      newSelectedTimeslots.splice(existentTimeslotIndex, 1);
+    }
+    else {
+      newSelectedTimeslots.push(newTimeslot);
+    }
+
+    if (newSelectedTimeslots.length > maxTimeslots) {
+      newSelectedTimeslots.splice(0, 1);
+    }
+
     this.setState({
-      selectedTimeslot: timeslot.utc(),
+      selectedTimeslots: newSelectedTimeslots,
     });
   }
 
 }
 
+Calendar.defaultProps = {
+  maxTimeslots: 1,
+  inputProps: {
+    names: {},
+  },
+};
+
 /**
- * @type {String} initialDate The initial date in which to place the calendar. Must be MomentJS parseable.
- * @type {Array} timeslots An array of timeslots to be displayed in each day.
- * @type {string} Initial value for timeslot input.
+ * @type {String} initialDate:  The initial date in which to place the calendar. Must be MomentJS parseable.
+ * @type {Array} timeslots:  An array of timeslots to be displayed in each day.
+ * @type {string} selectedTimeslot: Initial value for timeslot input.
+ * @type {Integer} maxTimexlots: maximum ammount of timeslots to select.
+ * @type {Object} inputProps: properties for the Inputs.
```

Comment:
Add better comments to `inputProps`. Since 2 names are required, this should be pointed at somewhere. You could even do:

`startDateInputProps` and `endDateInputProps`. That'd be even better